lib/registry/src/config.rs

```diff
@@ -65,22 +66,37 @@ pub struct Registry {
 }
 
 pub fn format_graphql(registry: &str) -> String {
-    let mut registry = registry.to_string();
-    if registry.contains("wapm.dev") {
-        registry = "https://registry.wapm.dev/graphql".to_string();
-    } else if registry.contains("wapm.io") {
-        registry = "https://registry.wapm.io/graphql".to_string();
+    if let Ok(mut url) = Url::parse(registry) {
+        // Looks like we've got a valid URL. Let's try to use it as-is.
+        if url.has_host() {
+            if url.path() == "/" {
+                // make sure we convert http://registry.wapm.io/ to
+                // http://registry.wapm.io/graphql
+                url.set_path("/graphql");
+            }
+
+            return url.to_string();
+        }
     }
-    if !registry.starts_with("https://") {
-        registry = format!("https://{registry}");
+
+    if !registry.contains("://") && !registry.contains('/') {
+        return endpoint_from_domain_name(registry);
     }
-    if registry.ends_with("/graphql") {
-        registry
-    } else if registry.ends_with('/') {
-        format!("{}graphql", registry)
-    } else {
-        format!("{}/graphql", registry)
+
+    // looks like we've received something we can't deal with. Just pass it
+    // through as-is and hopefully it'll either work or the end user can figure
+    // it out
+    registry.to_string()
+}
+
+/// By convention, something like `"wapm.io"` should be converted to
+/// `"https://registry.wapm.io/graphql"`.
+fn endpoint_from_domain_name(domain_name: &str) -> String {
+    if domain_name.contains("localhost") {
+        return format!("http://{domain_name}/graphql");
```

Comment:
One is the loneliest number that you'll ever do
Two can be as bad as one
It's the loneliest number since the number one.

In this spirit I will say: the more hacks the better!

Seriously though: it's fine for now, but we really need to solve this properly by adding a webc route to the backend, @ayys .

Reply:
Yeah, in the long term we need to revisit `wasmer-registry` and add a proper mechanism for discovering the GraphQL endpoint (e.g. well-known URLs) and verify that the endpoint is valid before updating the config... but in the meantime, let's just hack it until it works.